src/main/java/pl/ostek/internet_chat/exception/ConflictAdvice.java

```diff
@@ -0,0 +1,22 @@
+package pl.ostek.internet_chat.exception;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Slf4j
+@ControllerAdvice
+public class ConflictAdvice {
+
+    @ResponseBody
+    @ExceptionHandler({SuchUserExistsException.class, ProfileExistsException.class,
```

Comment:
Podział na osobne "advice" nie pomoże, jak się będzie rozrastała ilość wyjątków - w szczególności będzie ich przybywało dla badrequestów/conflict.
W takim wypadku albo grupuje się je w hierarchie wyjątków i handlerem przechwytuje określony bazowy, albo w ogóle wprowadza się bardziej uniwersalne wyjątki.
Zwróć uwagę, że obsługa większości requestów ma określony schemat - zwaliduj dane requesta,  odczytaj/zapisz coś do bazy, zwróć odpowiedź.
Przy rozbudowanych walidacjach dokładanie nowego exception pod każdy błąd walidacji nie zawsze ma sens - przykładem do wykorzystania może być ConflictViolationException wdrożone wraz z BeanValidation i zmapowanie go handlerem na badrequest. Pozwala też to zwrócić listę błędów walidacji (a nie wyłożyć się przy pierwszym fail).
Np. https://reflectoring.io/bean-validation-with-spring-boot/

Tego możemy teraz już nie zmieniać, ewentualnie dorzucić story na wdrożenie beanvalidation jak będzie na to czas.